cdap-docs/_common/_source/index.rst

```diff
@@ -106,6 +106,12 @@ on the installation, monitoring and diagnosing fully distributed CDAP in a Hadoo
 - |faq|_ Answers to questions you might have about CDAP
 
 
+.. |search| replace:: **Search:**
+.. _search: search.html
+
+- |search|_ Search this documentation, using either *Quick Search* or *Google*
```

Comment:
No, we have both. Google search does the site-wide; Quick Search restricts itself to the doc set, but is not as extensive. Both are implemented with a simple HTML form, such as
````
.. raw:: html

  <form action="http://www.google.com/search" name="searchbox" method="get" /> 
    <input type="hidden" name="hl" value="en" /> 
    <input type="hidden" name="ie" value="ISO-8859-1" /> 
    <input type="hidden" name="sitesearch" value="docs.cask.co/cdap/" /> 
    <input maxlength="256" size="40" name="q" value="" /> 
    <input type="submit" value="Search using Google" name="btnG" 
      style="font-size:75%;" /> 
  </form>
````